sympy/sets/sets.py

```diff
@@ -1879,6 +1873,27 @@ def as_relational(self, symbol):
 
         return Xor(A_rel, B_rel)
 
+    @property
+    def is_iterable(self):
+        if all(arg.is_iterable for arg in self.args):
+            return True
+
+    def __iter__(self):
+        if not self.is_iterable:
+            raise TypeError("{} is not iterable".format(self))
+
+        args = self.args
+        union = roundrobin(*(iter(arg) for arg in args))
+
+        for item in union:
+            count = 0
+            for s in args:
+                if item in s:
+                    count += 1
```

Comment:
This can also raise. We have `._contains` that returns `None` instead of raising but what will you yield of if you get `None`?